### PR TITLE
PT-50: Update Generic Crud SearchService

### PR DESCRIPTION
**Pull request by aizhuli**
## Description
Update generic crud search service by including post-read processing of search results since additional processing logic is required by implementations  in modules

### CODE REVIEW

**Review by akak1977 — APPROVED** [review]

File: src/VirtoCommerce.Platform.Data/GenericCrud/CrudService.cs [hunk 1]
```diff
@@ -257,11 +257,25 @@ public virtual async Task DeleteAsync(IEnumerable<string> ids, bool softDelete =
...
             }
         }
 
+        /// <summary>
+        /// Custom CRUD service can override to implement some actions after delete
+        /// </summary>
+        /// <param name="models"></param>
+        /// <param name="changedEntries"></param>
+        /// <returns></returns>
+        protected virtual Task AfterDeleteAsync(IEnumerable<TModel> models, IEnumerable<GenericChangedEntry<TModel>> changedEntries)
```
Comment (akak1977) [inline src/VirtoCommerce.Platform.Data/GenericCrud/CrudService.cs:273]: Does it really need?
Why not just override _Delete_, then call _base.Delete_ inside of, then do your custom actions.

**Review by akak1977 — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Platform.Data/GenericCrud/CrudService.cs (hunk 1 above)
Comment (aizhuli) [reply src/VirtoCommerce.Platform.Data/GenericCrud/CrudService.cs:273]: If entry changed events need to be published only after all necessary actions are done, we need this method.

Comment (akak1977) [reply src/VirtoCommerce.Platform.Data/GenericCrud/CrudService.cs:273]: Ок

**Review by akak1977 — APPROVED** [review]

**Comment by mvktsk** [comment]
Actual PR https://github.com/VirtoCommerce/vc-platform/pull/2363

**Pull request closed without merging**